### PR TITLE
Added binary to octal conversion

### PR DESCRIPTION
**Pull request by VishnuPothan**
# Welcome to Dart community

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: conversions/binary_to_octal.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
+//Author:VishnuPothan
```
Comment (Parowicz) [inline conversions/binary_to_octal.dart:1]: You could add some tests
```suggestion
import 'package:test/test.dart';

//Author:VishnuPothan
```

File: conversions/binary_to_octal.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+//Binary number to octal number conversion
+void main() {
+  print(binary_to_octal("-1111")); // -17
+  print(binary_to_octal("101011")); // 53
+  print(binary_to_octal("1011a01")); // error
+  print(binary_to_octal("")); // error
```
Comment (Parowicz) [inline conversions/binary_to_octal.dart:7]:
```suggestion
  test("binary_to_octal -1111", () {
    expect(binary_to_octal("-1111"), equals("-17"));
  });

  test("binary_to_octal 101011", () {
    expect(binary_to_octal("101011"), equals("53"));
  });

  test("binary_to_octal rasies error when number is invalid", () {
    expect(() => binary_to_octal("-1011a01"), throwsFormatException);
  });

  test("binary_to_octal of empty string raises error", () {
    expect(() => binary_to_octal(""), throwsFormatException);
  });
```

File: conversions/binary_to_octal.dart [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+String binary_to_octal(String bin_string) {
+  bin_string = bin_string.trim();
+  if (bin_string == null || bin_string == "") {
+    return ("An empty value was passed to the function");
```
Comment (Parowicz) [inline conversions/binary_to_octal.dart:13]: Throwing exception is more intuitive here
```suggestion
    throw new FormatException("An empty value was passed to the function");
```

File: conversions/binary_to_octal.dart [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  bool is_negative = bin_string[0] == "-";
+  if (is_negative) bin_string = bin_string.substring(1);
+
+  String octal_val = "";
+  int binary;
+  try {
+    binary = int.parse(bin_string);
+  } catch (e) {
+    return ("An invalid value was passed to the function");
```
Comment (Parowicz) [inline conversions/binary_to_octal.dart:23]: Throwing exception is more intuitive here
```suggestion
    throw new FormatException("An invalid value was passed to the function");
```

**Comment by VishnuPothan** [comment]
@Parowicz added the test cases

**Comment by Parowicz** [comment]
Thank you! Looks great!

**Pull request merged**